File: typescript/src/models/total.ts
// this file is @generated
import { type TotalType, TotalTypeSerializer } from "./totalType";

export interface Total {
  type: TotalType;
  displayText: string;
  amount: number;
}

export const TotalSerializer = {
  _fromJsonObject(object: any): Total {
    return {
      type: TotalTypeSerializer._fromJsonObject(object["type"]),
      displayText: object["display_text"],
      amount: object["amount"],
    };
  },

  _toJsonObject(self: Total): any {
    return {
      type: TotalTypeSerializer._toJsonObject(self.type),
      display_text: self.displayText,
      amount: self.amount,
    };
  },
};

File: rust/src/models/total.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::total_type::TotalType;

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct Total {
    pub r#type: TotalType,
    pub display_text: String,
    pub amount: i32,
}
